### PR TITLE
Add conda recipe for cgns v3.2.1

### PR DESCRIPTION
**Pull request by mdpiper**
This PR includes a recipe for the [CGNS](https://github.com/CGNS/CGNS/) library, version 3.2.1. This is an older version of the library, but this particular version is needed as a dependency for other software.

The text of the license file is pulled directly from the [README](https://github.com/CGNS/CGNS/blob/develop/README.md) of the CGNS GitHub repository.

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mdpiper** [comment]
Hi, @brtnfld -- I just wanted to let you know that I'm building a [conda-forge](https://conda-forge.org/) recipe for CGNS v3.2.1. I realize this is an older version of CGNS, but it's the version being used in the USGS [FaSTMECH](https://i-ric.org/en/solvers/fastmech/) river flow model. If it would serve your community, I can also create recipes for newer versions of CGNS -- I have one mostly prepared for v3.3.1, and I can do one for v3.4.0.

File: recipes/cgns/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  skip: True  # [win]
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('fortran') }}
+    - cmake
+    - make
+    - zlib
+    - hdf5 =1.8
```
Comment (scopatz) [inline recipes/cgns/meta.yaml]: zlib and hdf5 should be host requirements

File: recipes/cgns/build.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#! /usr/bin/env bash
+
+_build_shared=ON
+if [[ `uname -s` == 'Darwin' ]]; then
+    export MACOSX_DEPLOYMENT_TARGET=""
```
Comment (scopatz) [inline recipes/cgns/build.sh]: it seems like you should not reset this

File: recipes/cgns/meta.yaml (hunk 1 above)
Comment (brtnfld) [reply recipes/cgns/meta.yaml]: FYI, if you don't build hdf5 with zlib then it is not required by CGNS. CGNS does not have compression (via HDF5).

File: recipes/cgns/build.sh (hunk 2 above)
Comment (brtnfld) [reply recipes/cgns/build.sh]: v3.4.0 would be nice, and built with hdf5 1.10. The difference in terms of build between 3.3 and 3.4 should be minimal.

Thanks.

**Comment by mdpiper** [comment]
@scopatz Thanks for reviewing this PR. I'll make the changes you've suggested and push them here.

**Comment by mdpiper** [comment]
@brtnfld Excellent! I'll work on v3.4.0 next.

File: recipes/cgns/meta.yaml (hunk 1 above)
Comment (mdpiper) [reply recipes/cgns/meta.yaml]: Fixed with 439d212.

Comment (mdpiper) [reply recipes/cgns/meta.yaml]: @brtnfld Thanks for the suggestion. I think conda-forge's hdf5 builds with zlib enabled -- here's its [build script](https://github.com/conda-forge/hdf5-feedstock/blob/master/recipe/build.sh).

File: recipes/cgns/build.sh (hunk 2 above)
Comment (mdpiper) [reply recipes/cgns/build.sh]: Fixed with 7f4f8f8.

**Comment by mdpiper** [comment]
@scopatz I think this PR is ready to go. Please let me know if you have any other suggestions.

**Comment by scopatz** [comment]
Thanks @mdpiper!

**Pull request merged**